pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -258,6 +258,7 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule) *envoy_jwt.JwtAuthenti
 			jwtPubKey, err = model.GetJwtKeyResolver().GetPublicKey(jwtRule.JwksUri)
 			if err != nil {
 				log.Errorf("Failed to fetch jwt public key from %q: %s", jwtRule.JwksUri, err)
+				jwtPubKey = `{"keys":[ {"e":"AQAB","kid":"abc","kty":"RSA","n":"failedToFetchJwtPubKey"}]}`
```

Comment:
Hey @xulingqing could you make "n" be dynamic (not always same default value) and also more complex?
I'm afraid of an attacker being able to generate key material from the base64 decoding of the relatively some number that the modulo "n" would turn once "failedToFetchJwtPubKey" would be base64 decoded.

Reply:
Thanks Francois! To make it more complex, you can encode the jwtRule.JwksUri or even the err in it, this also gives a nicer debugging message.

Reply:
We should not make it dynamic on every push or it will result in draining in envoy. But agree with @fpesce in general

Reply:
it should be the same on every push if we only include JwksUri

Reply:
Sure, Thanks for pointing it out, Francois. I'm working on it.

Reply:
Maybe in the future we can add salt (like. a random number with fixed seed)

Reply:
Hi Francois @fpesce and Yangmin @yangminzhu , I've tried to make the jwks dynamic by directly append issuer or jwksURI. But those none letter character like "/" will influence the parsing of jwks. Error logs is like:
```
2020-07-31T15:29:12.362507Z     warning envoy config    [external/envoy/source/common/config/grpc_subscription_impl.cc:101] gRPC config for type.googleapis.com/envoy.api.v2.Listener rejected: Error adding/updating listener(s) virtualInbound: Issuer 'test-issuer-1@istio.io' in jwt_authn config has invalid local jwks: Jwks RSA [n] or [e] field is missing or has a parse error
```

Here are some methods I'm thinking of to make it dynamic.
1. extract letters from policy Name or Issuer field
2. encode any field to base64 and append

Do you have any suggestion?

Reply:
base64 encoding seems the best (as this field is ultimately base64 decoded). You can add random content in it too.

Reply:
Can we use base64 here, you can add an extra `_` between `failedToFetchJwksFor` and the base64 string to make it a little more readable.